fahrschein-spring-boot-starter/src/main/java/org/zalando/spring/boot/fahrschein/nakadi/config/FahrscheinRegistrar.java

```diff
@@ -65,15 +66,16 @@ private String registerPublisher(PublisherConfig publisherConfig) {
         });
     }
 
-    private String registerNakadiClient(AbstractConfig consumerConfig, String type) {
-        return registry.registerIfAbsent(consumerConfig.getId() + "-" + type, NakadiClient.class, () -> {
-            log.info(LOG_PREFIX + "NakadiClient ...", consumerConfig.getId());
-            final String requestFactoryRef = registerRequestFactory(consumerConfig);
+    private String registerNakadiClient(AbstractConfig config, String type) {
```

Comment:
The same method is used for registering the Nakadi client by both consumers and publishers, but we only require instrumentation for publishers. Perhaps it's time to consider creating dedicated clients for each role – one for consumers and one for publishers.

Reply:
agree, but too big of a breaking change for now.